ee/zentral/contrib/intune/models.py

```diff
@@ -35,6 +35,18 @@ def save(self, *args, **kwargs):
             self.version = F("version") + 1
         super().save(*args, **kwargs)
 
+    def serialize_for_event(self):
+        return {
+            'business_unit': self.business_unit.pk,
+            'name': self.name,
+            'description': self.description,
+            'tenant_id': self.tenant_id,
+            'client_id': str(self.client_id),
+            'client_secret': self.client_secret,
```

Comment:
It is good to have the encrypted client secret in the event, because otherwise we could not see the different when the client secret is updated. **BUT** the default secret encryption in Zentral is … not an encryption, it is just a base64 encoding. So, I think it would be better to replace the `client_secret` member with a `client_secret_hash` and use hashlib.sha256 on the decrypted client secret. This way 1) it is easy for the users to compare with the value they know, 2) we never leak the secret, even if the secret engine is not configured in Zentral. Also, the secret is generated by Microsoft and is good enough to not have to salt the hash I think…